chain/xrp/client/client.go

```diff
@@ -0,0 +1,992 @@
+						contract = node.ModifiedNode.FinalFields.HighLimit.Issuer
+					} else {
+						transactionType = "destination"
+						contract = node.ModifiedNode.FinalFields.LowLimit.Issuer
+					}
+				}
+
+				convertedTransactedAmount, conversionErr := extractModifiedNodeBalance(node.ModifiedNode)
+				if conversionErr != nil {
+					return xclient.TxInfo{}, conversionErr
+				}
+
+				//if modifiedNode.FinalFields.Account != txResponse.Result.Account {
+				if transactionType == "destination" {
+					tf.AddDestination(
+						xc.Address(modifiedNode.FinalFields.LowLimit.Issuer),
+						xc.ContractAddress(contract),
+						convertedTransactedAmount,
+						nil,
+					)
+				} else if transactionType == "source" {
+					tf.AddSource(
+						xc.Address(modifiedNode.FinalFields.HighLimit.Issuer),
+						xc.ContractAddress(contract),
+						convertedTransactedAmount,
+						nil,
+					)
+				}
+			}
+
+		} else if node.CreatedNode != nil {
+
+			createdNode := node.CreatedNode
+			if createdNode.LedgerEntryType == "AccountRoot" {
+
+				var (
+					contract   string
+					newBalance float64
+					decimals   int32
+					//transactionType string
+					parseErr error
+				)
+
+				if node.CreatedNode.NewFields.Balance.XRPAmount != "" {
+					contract = ""
+					decimals = 0
+					newBalance, parseErr = strconv.ParseFloat(node.CreatedNode.NewFields.Balance.XRPAmount, 10)
+					if parseErr != nil {
+						return xclient.TxInfo{}, parseErr
+					}
+
+				} else {
+					decimals = 15
+					newBalance, parseErr = strconv.ParseFloat(node.CreatedNode.NewFields.Balance.TokenAmount.Value, 10)
+					if parseErr != nil {
+						return xclient.TxInfo{}, parseErr
+					}
+
+					if newBalance > 0 {
+						//transactionType = "source"
+						contract = node.CreatedNode.NewFields.HighLimit.Issuer
+					} else {
+						//transactionType = "destination"
+						contract = node.CreatedNode.NewFields.LowLimit.Issuer
+					}
+				}
+
+				var newBalanceString string
+				if decimals == 0 {
+					newBalanceString = fmt.Sprintf("%.6f", math.Abs(newBalance))
+				} else {
+					newBalanceString = fmt.Sprintf("%.15f", math.Abs(newBalance))
+				}
+
+				newBalanceString = strings.TrimRight(strings.TrimRight(newBalanceString, "0"), ".")
+
+				convertedTransactedAmount, conversionErr := xc.NewAmountHumanReadableFromStr(newBalanceString)
+				if conversionErr != nil {
+					return xclient.TxInfo{}, conversionErr
+				}
+
+				//if transactionType == "destination" {
+				if createdNode.NewFields.Account != "" {
+					tf.AddDestination(
+						xc.Address(createdNode.NewFields.Account),
+						xc.ContractAddress(contract),
+						convertedTransactedAmount.ToBlockchain(decimals),
+						nil,
+					)
+				} else {
+					tf.AddDestination(
+						xc.Address(createdNode.NewFields.LowLimit.Issuer),
+						xc.ContractAddress(contract),
+						convertedTransactedAmount.ToBlockchain(decimals),
+						nil,
+					)
+				}
+
+			} else if createdNode.LedgerEntryType == "RippleState" {
```

Comment:
I think this also has the same bug.  If `transactionType == ""` like when XRP is transferred, then no transfer gets updated.  Please refactor to not need this logic, or use bool.